Firestore/core/src/util/byte_stream_apple.mm

```diff
@@ -42,45 +42,33 @@
     }
   }
 
-  // One last try since the loop might break because eof or max_length reached.
-  if (found_at == std::string::npos) {
-    found_at = buffer_.find(delim);
-  }
-
   // Still not found, return the whole `buffer_` and clear it.
   if (found_at == std::string::npos) {
-    auto read_result = StreamReadResult(std::move(buffer_), eof());
-    buffer_.clear();
-    return read_result;
+    return ConsumeBuffer();
   }
 
   // Found, return the proper substring and erase the substring.
   std::string result = buffer_.substr(0, found_at);
   buffer_.erase(0, found_at);
-  auto read_result = StreamReadResult(std::move(result), eof());
+  StreamReadResult read_result(std::move(result), eof());
   return read_result;
 }
 
 StreamReadResult ByteStreamApple::Read(size_t max_length) {
   // Serve from buffer_
-  if (buffer_.size() >= max_length) {
+  if (buffer_.size() > max_length) {
```

Comment:
Was this caught by tests?

Reply:
Yes: `ReadsFromInternalBuffer_AppleImpl`